kotlin-code-style.md

```diff
@@ -40,49 +29,43 @@ class Person(val firstName: String, val lastName: String, var age: Int) {
 }
 ```
 
-# Extensions
+## Constructors
 
-When adding extensions to external classes, make separate files for each type
-and name them _<Type>Extensions.kt_:
+As often as possible we should specify if the constructor properties are mutable
+(`var`) or read-only (`val`):
 
-```
-- StringExtensions.kt
-- BitmapExtensions.kt
-- ContextExtensions.kt
-- ...
+```kotlin
+class Person(val firstName: String, val lastName: String, var age: Int) {
+    // ...
+}
 ```
 
-# Object Expressions and Declarations
+## Methods
 
-## Object expressions
-
-To avoid writing methods that are too long, externalize object expressions as
-properties:
+If your method has an empty body, use the `Unit` type instead of empty bracket body : 
 
 ```kotlin
-class Person {
-
-    private val anAttribute = object : MyJavaInterface() {
-        // ...
-    }
-
-    fun aMethod() {
-        button.setOnSomeClick(anAttribute)
-    }
+fun anEmptyFunction() = Unit
+```
 
+if you are implementing an interface, and only one method is relevant for you, place all the empty one first :
+```kotlin
+object : someInterface {
+  fun methodOne() = Unit
+  fun methodThree() = Unit
+  fun methodTwo() {
+    //your code here
+  }
 }
 ```
 
-# Delegated Properties
-
-## Standard Delegates
+# Extensions
 
-We should use the standard delegates `lazy` as often as possible in order to
-have lazy loading properties:
+When adding extensions to external classes, create a extension package and make separate files for each type:
 
-```kotlin
-val myValue : String by lazy {
-    someInit()
-    functionReturningAStringValue()
-}
+```
+- extensions
+  - String.kt
+  - Bitmap.kt
+  - ...
```

Comment:
I kind of still prefer the filenames to be _StringExtensions.kt_, _BitmapExtensions.kt_, etc. That's more convenient to open from the fuzzy-finder file opener thingy as well, in my opinion.